sched/pthread/pthread_detach.c

```diff
@@ -96,12 +96,15 @@ int pthread_detach(pthread_t thread)
            * thread exits
            */
 
-          pjoin->detached = true;
+          if (pjoin->detached)
+            {
+              ret = EINVAL;
```

Comment:
I think that standard does not define this behavior. Are you trying to bring Linux behavior here? I mean what is the motivation of this change?

Reply:
I mean `The behavior is undefined if the value specified by the thread argument to pthread_detach() does not refer to a joinable thread.`